### PR TITLE
Simple unstash in a different directory example

### PR DESCRIPTION
**Pull request by abayer**
cc @reviewbybees

### CODE REVIEW

**Comment by rtyler** [comment]
:beer:

**Comment by amuniz** [comment]
:beer: and :bee: :)

File: unstash-different-dir/unstashDifferentDir.groovy [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// Run on a node with the "first-node" label.
+node('first-node') {
+    // Make the output directory.
+    sh "mkdir -p output"
+
+    // Write a text file there.
+    writeFile file: "output/somefile", text: "Hey look, some text."
+
+    // Stash that directory and file.
+    stash name: "first-stash", includes: "output/*"
```
Comment (jglick) [inline unstash-different-dir/unstashDifferentDir.groovy]: Can use `output/` (same as `output/**`, i.e., all files under the dir).

File: unstash-different-dir/unstashDifferentDir.groovy [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+// Next, we'll make a new directory on a second node, and unstash the original
+// into that new directory, rather than into the root of the biuld.
+stage "second step on second node"
+
+// Run on a node with the "second-node" label.
+node('second-node') {
+    // Make the directory we're unstashing to.
+    sh "mkdir -p first-stash"
```
Comment (jglick) [inline unstash-different-dir/unstashDifferentDir.groovy]: I _think_ this can be skipped.

File: unstash-different-dir/unstashDifferentDir.groovy [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    // Run the unstash from within that directory!
+    dir("first-stash") {
+        unstash "first-stash"
+    }
+
+    // Look, no output directory under the root!
+    sh "ls -la ${pwd()}"
```
Comment (jglick) [inline unstash-different-dir/unstashDifferentDir.groovy]: Just `ls -la .` suffices.

File: unstash-different-dir/unstashDifferentDir.groovy [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    // And look, output directory is there under first-stash!
+    sh "ls -la ${pwd()}/first-stash"
```
Comment (jglick) [inline unstash-different-dir/unstashDifferentDir.groovy]: Similarly suffices to run `ls -la first-stash`.

**Comment by jglick** [comment]
:bee:

File: unstash-different-dir/unstashDifferentDir.groovy (hunk 1 above)
Comment (abayer) [inline unstash-different-dir/unstashDifferentDir.groovy]: I'll add a note of the various formats...

File: unstash-different-dir/unstashDifferentDir.groovy (hunk 2 above)
Comment (abayer) [inline unstash-different-dir/unstashDifferentDir.groovy]: I tend to do mkdir -p even when I probably don't need to - but I'll think about tweaking this.

File: unstash-different-dir/unstashDifferentDir.groovy (hunk 3 above)
Comment (abayer) [inline unstash-different-dir/unstashDifferentDir.groovy]: I'm deliberately using pwd() 'cos I'm trying to educate on that as well. =)

**Pull request merged**